docs/concepts/models.md

```diff
@@ -648,62 +648,96 @@ Here are some additional notes on the behavior of [`model_construct()`][pydantic
-3. Use the type variables as annotations where you will want to replace them with other types.
+=== "Python 3.8 and above"
 
-!!! warning "PEP 695 support"
-    Pydantic does not support the new syntax for generic classes (introduced by [PEP 695](https://peps.python.org/pep-0695/)),
-    available since Python 3.12. Progress can be tracked in [this issue](https://github.com/pydantic/pydantic/issues/9782).
+    ```python {upgrade="skip"}
+    from typing import Generic, TypeVar
 
-Here is an example using a generic Pydantic model to create an easily-reused HTTP response payload wrapper:
+    from pydantic import BaseModel, ValidationError
 
-```python
-from typing import Generic, List, Optional, TypeVar
+    DataT = TypeVar('DataT')  # (1)!
 
-from pydantic import BaseModel, ValidationError
 
-DataT = TypeVar('DataT')  # (1)!
+    class DataModel(BaseModel):
+        number: int
 
 
-class DataModel(BaseModel):
-    numbers: List[int]
-    people: List[str]
+    class Response(BaseModel, Generic[DataT]):  # (2)!
+        data: DataT  # (3)!
 
 
-class Response(BaseModel, Generic[DataT]):  # (2)!
-    data: Optional[DataT] = None  # (3)!
+    print(Response[int](data=1))
+    #> data=1
+    print(Response[str](data='value'))
+    #> data='value'
+    print(Response[str](data='value').model_dump())
+    #> {'data': 'value'}
 
+    data = DataModel(number=1)
+    print(Response[DataModel](data=data).model_dump())
+    #> {'data': {'number': 1}}
+    try:
+        Response[int](data='value')
+    except ValidationError as e:
+        print(e)
+        """
+        1 validation error for Response[int]
+        data
+          Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='value', input_type=str]
+        """
+    ```
 
-print(Response[int](data=1))
-#> data=1
-print(Response[str](data='value'))
-#> data='value'
-print(Response[str](data='value').model_dump())
-#> {'data': 'value'}
+    1. Declare one or more [type variables][typing.TypeVar] to use to parameterize your model.
+    2. Declare a Pydantic model that inherits from [`BaseModel`][pydantic.BaseModel] and [`typing.Generic`][]
+       (in this specific order), and add the list of type variables you declared previously as parameters to the
+       [`Generic`][typing.Generic] parent.
+    3. Use the type variables as annotations where you will want to replace them with other types.
 
-data = DataModel(numbers=[1, 2, 3], people=[])
-print(Response[DataModel](data=data).model_dump())
-#> {'data': {'numbers': [1, 2, 3], 'people': []}}
-try:
-    Response[int](data='value')
-except ValidationError as e:
-    print(e)
-    """
-    1 validation error for Response[int]
-    data
-      Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='value', input_type=str]
-    """
-```
+=== "Python 3.12 and above (new syntax)"
```

Comment:
```suggestion
=== "Python 3.12+ (new type parameter syntax)"
```

Reply:
I used the same wording as our upgrade plugin.

Otherwise, changed to _new type parameter syntax_